### PR TITLE
Add a config option to allow users to enable post quantum TLS support in CRT HTTP client

### PR DESCRIPTION
**Pull request by zoewangg**
## Context
TlsPreference API was removed as part of https://github.com/aws/aws-sdk-java-v2/pull/3706. This PR adds support back through a different API

## Modifications
Add a config option to allow users to enable post quantum TLS support in CRT HTTP client

## Testing
Added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java [hunk 1]
```diff
@@ -111,67 +115,10 @@ private AwsCrtAsyncHttpClient(DefaultBuilder builder, AttributeMap config) {
             this.maxConnectionsPerEndpoint = config.get(SdkHttpConfigurationOption.MAX_CONNECTIONS);
             this.monitoringOptions = revolveHttpMonitoringOptions(builder.connectionHealthConfiguration);
             this.maxConnectionIdleInMilliseconds = config.get(SdkHttpConfigurationOption.CONNECTION_MAX_IDLE_TIMEOUT).toMillis();
-            this.proxyOptions = buildProxyOptions(builder.proxyConfiguration);
+            this.proxyOptions = buildProxyOptions(builder.proxyConfiguration, tlsContext);
         }
     }
 
-    private HttpMonitoringOptions revolveHttpMonitoringOptions(ConnectionHealthConfiguration config) {
```
Comment (zoewangg) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:118]: No changes have been made to these methods. I just moved them to the new class.

**Review by alexw91 — APPROVED** [review]

File: .changes/next-release/feature-AWSCRTHTTPClient-5e6e380.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "type": "feature",
+    "category": "AWS CRT HTTP Client",
+    "contributor": "",
+    "description": "Allow users to enable Post Quantum TLS via `AwsCrtAsyncHttpClient.builder().preferePostQuantumTls(true).build()`."
```
Comment (cenedhryn) [inline .changes/next-release/feature-AWSCRTHTTPClient-5e6e380.json:5]: Misspelling in `AwsCrtAsyncHttpClient.builder().preferePostQuantumTls(true).build()`. Should be `prefer`

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            clientProxyOptions.setAuthorizationPassword(proxyConfiguration.password());
+            clientProxyOptions.setAuthorizationType(HttpProxyOptions.HttpProxyAuthorizationType.Basic);
+        } else {
+            clientProxyOptions.setAuthorizationType(HttpProxyOptions.HttpProxyAuthorizationType.None);
+        }
+
+        return clientProxyOptions;
+    }
+
+    public static HttpMonitoringOptions revolveHttpMonitoringOptions(ConnectionHealthConfiguration config) {
```
Comment (cenedhryn) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java:84]: Nit: there's an old misspelling here, `revolve` should be `resolve`. I think.

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java (hunk 1 above)
Comment (cenedhryn) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java:118]: gotcha

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java (hunk 3 above)
Comment (zoewangg) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java:84]: The person(me) who wrote this clearly can't spell!

**Review by cenedhryn — APPROVED** [review]

**Pull request merged**